libbindgen/src/ir/item.rs

```diff
@@ -525,6 +526,107 @@ impl Item {
         self.as_type().map_or(false, |ty| ty.is_type_ref())
     }
 
+    /// Get the target item id for name generation.
+    fn name_target(&self, ctx: &BindgenContext, for_name_checking: bool) -> ItemId {
+        let mut item = self;
+        loop {
+            match *item.kind() {
+                ItemKind::Type(ref ty) => {
+                    match *ty.kind() {
+                        // If we're a template specialization, our name is our
+                        // parent's name.
+                        TypeKind::Comp(ref ci) if ci.is_template_specialization() => {
+                            item = ctx.resolve_item(ci.specialized_template().unwrap());
```

Comment:
I'll do this, but also add a `debug_assert!` that the specialized template's name target is itself.